### ISSUE TITLE
Could not connect to "<speaker> (AirSonos)"

### ISSUE THREAD

**Issue by jkili** [stephen/airsonos#80]
![screen shot 2014-12-05 at 1 11 45 pm](https://cloud.githubusercontent.com/assets/3908477/5310006/655c039e-7c80-11e4-945e-113eaeb7b726.png)

airsonos works great 50% of the time (probably 100% on a fresh boot). However, I think if I have run airsonos previously and got to use it later (i.e. after sleep) it starts up normally but when I go to select the speaker it says 'Could not connect to "Living Room (AirSonos)"'. A reboot will fix. 

Is there anything I can do to make it work without the reboot?

-Justin

Here is the verbose startup output:
```
Searching for Sonos devices on network...
Setting up AirSonos for Living Room {192.168.1.14:1400}
  nodetunes:server starting nodetunes server (Living Room (AirSonos)) +0ms
  nodetunes:server broadcasting mdns advertisement (for port 5000) +8ms
Setting up AirSonos for Upstairs {192.168.1.16:1400}
  nodetunes:server starting nodetunes server (Upstairs (AirSonos)) +2ms
  nodetunes:server broadcasting mdns advertisement (for port 5001) +5ms
```

I get no stdout when I used the command: node examples/server_stdout.js and use the "NodeTunes" speaker but I get the same message.

Yosemite 10.10.1 and latest code (from today but has been happening since I started using airsonos months ago on Mavericks and Yosemite).

**Comment by Diode0** [comment]
I have the same problem as @adamburmister, i.e. I never get any audio working
I observe similar output to @celtboy and @adamburmister, please see below (I attempted playback from an iPad running iOS 8.0.2 and an iPhone running iOS 8.1.2, my airsonos is an install via npm on an up-to-date Raspbian):

pi@raspberrypi:~$ airsonos --verbose
*** WARNING *** The program 'node' uses the Apple Bonjour compatibility layer of Avahi.
*** WARNING *** Please fix your application to use the native API of Avahi!
*** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=node>
*** WARNING *** The program 'node' called 'DNSServiceRegister()' which is not supported (or only supported partially) in the Apple Bonjour compatibility layer of Avahi.
*** WARNING *** Please fix your application to use the native API of Avahi!
*** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=node&f=DNSServiceRegister>
Searching for Sonos devices on network...
Setting up AirSonos for Living Room {192.168.1.28:1400}
  nodetunes:server starting nodetunes server (Living Room (AirSonos)) +0ms
  nodetunes:server broadcasting mdns advertisement (for port 5000) +321ms
Setting up AirSonos for Portable {192.168.1.27:1400}
  nodetunes:server starting nodetunes server (Portable (AirSonos)) +97ms
  nodetunes:server broadcasting mdns advertisement (for port 5001) +240ms
  nodetunes:rtsp received method OPTIONS (CSeq: 0)
{ cseq: '0',
  'x-apple-device-id': '0x9c04ebf07660',
  'apple-challenge': 'Tp9s2SmvqpncqE4viogHOg==',
  'dacp-id': '168EF43D7A840CC7',
  'active-remote': '2782886928',
  'user-agent': 'AirPlay/210.98' } +19m
  nodetunes:helper building challenge for Tp9s2SmvqpncqE4viogHOg== (ip: c0a8011e, mac: 545200b07a7f) +8ms
  nodetunes:helper computed challenge: q2fJIsT9kSmgB1RxcGlsW6Y0pgIKjieAy/a5gNPMyC1J+V7vDeHSKtoqUaFPOHk6YyeruB7DRxhVjqqBOocJEbOX2tdP5RPTtJb1xXNLj6PXxx5HSKGv9lrroU4cutcsvNUrkSpTFWgIOcYfnMu+M8g3wWutEE/hz7lGzezRfQEOVTPkBZ9kXPKL+/CUrG4Q6+5BxiQaV0FAaVxKhNJblGQD0gd7xKraFIkyVkCOfHimbW2oyAOV1qNqJQJQjF0QU8ERYZpG298bxTeSlbql5pUAxwfBug9DytXptiSyhAY9JfK1g/41hvgwB0CS6OEqUf0i4r1OLAv8dE4DMV/muA== +12s
  nodetunes:rtsp received method OPTIONS (CSeq: 0)
{ cseq: '0',
  'x-apple-device-id': '0x649abe03b61f',
  'apple-challenge': 'AxRkP7cSV1GliCWbudrT+A==',
  'dacp-id': '9BF973FE77CCA25E',
  'active-remote': '537031523',
  'user-agent': 'AirPlay/211.3' } +32s
  nodetunes:helper building challenge for AxRkP7cSV1GliCWbudrT+A== (ip: c0a8011e, mac: 545200b07a7f) +4ms
  nodetunes:helper computed challenge: Cgwlm9lnGm4hOkreplOSRy2yO/+QZv6ASkFCfjlSwPnZPiTXx1OBlWCje3Gs53dLg4asTBl3nvhFOOn7AJsyWXra+u+g2z71jdCM650uoInz+u9K6t6eQuB7MjQUhs/ZRIlCmvoXYThIq28RRkWokhlFy9dBjB++BUVpsqT+5azLHaD2xyuKJuA9csHYjVCGInig3x/Gud3KrWID13M5SAre+wRRO6LidzpYm3zBcyyz3icpfXYJ9Su5W1WpclJmTnVs07kCcdTAlcVLQokOGjNu17UMbmsg47juU5K8RLSIaB1T0riXEjbKjV5Y121Oe3YrVUUbNEPVh9hr36WnCA== +12s

**Comment by stephen** [comment]
@celtboy and @Diode0 - can you verify the network IPs for the machines you're running airsonos on? They're reported as `192.168.2.6`, and `192.168.1.30`, respectively (acc. the dump).

I'm wondering if the challenge is not being accepted because the device thinks the IP is wrong..

@jkili - this might actually be a Yosemite/AirPlay stability issue. Instead of restarting when this happens, try running `sudo killall codeaudiod`. I know that's a crappy solution, but in those cases (I believe) it's actually Yosemite dropping the ball.

**Comment by Diode0** [comment]
Thanks @stephen for the follow-up

Yes, the IP address in the dump is correct:

pi@raspberrypi:~$ ifconfig
eth0      Link encap:Ethernet  HWaddr b8:27:eb:6e:fa:0c  
          inet addr:192.168.1.30  Bcast:192.168.1.255  Mask:255.255.255.0
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:394030 errors:0 dropped:1 overruns:0 frame:0
          TX packets:16442 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:44123451 (42.0 MiB)  TX bytes:2651480 (2.5 MiB)

**Comment by celtboy** [comment]
@stephen also thank you for the follow-up.  Just chiming in to confirm my
address is the same as well.

**Comment by benzwu** [comment]
Mine didn't resolve with reboot too

**Comment by jkili** [comment]
Thanks Stephen, I will give it a go. Cheers

**Comment by celtboy** [comment]
Why did you close this issue? It's still very much alive.

**Comment by celtboy** [comment]
Thank you, Stephen!

**Comment by iDuuck** [comment]
Mine is showing the wrong IP .. my ip is .0.15 but the client shows .0.11

**Comment by catesandrew** [comment]
interesting, I have airsonos running on 10.7 (lion) and haven't seen any issues so it might be related to yosemite dropping the ball. will confirm with walk through though. also I'm running it through supervisor so its possible supervisor is restarting it unbeknownst to me.

**Comment by Diode0** [comment]
Hello Stephen

I apt-get upgraded my raspberry pi, rebooted it and then, upgraded node to 0.10.36 and airsonos to 0.1.1.

Unfortunately, I still cannot get audio working at all and thus, this issue remains valid.

Thanks in advance for looking into it.

Please see below updated verbose logs as well as diagnostics output:

pi@raspberrypi:/tmp$ airsonos --verbose
*** WARNING *** The program 'node' uses the Apple Bonjour compatibility layer of Avahi.
*** WARNING *** Please fix your application to use the native API of Avahi!
*** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=node>
*** WARNING *** The program 'node' called 'DNSServiceRegister()' which is not supported (or only supported partially) in the Apple Bonjour compatibility layer of Avahi.
*** WARNING *** Please fix your application to use the native API of Avahi!
*** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=node&f=DNSServiceRegister>
Searching for Sonos devices on network...
Setting up AirSonos for Portable {192.168.1.27:1400}
  nodetunes:server starting nodetunes server (Portable (AirSonos)) +0ms
  nodetunes:server broadcasting mdns advertisement (for port 5000) +308ms
Setting up AirSonos for Portable {192.168.1.27:1400}
  nodetunes:server starting nodetunes server (Portable (AirSonos)) +73ms
  nodetunes:server broadcasting mdns advertisement (for port 5001) +245ms
Setting up AirSonos for Living Room {192.168.1.28:1400}
  nodetunes:server starting nodetunes server (Living Room (AirSonos)) +50ms
  nodetunes:server broadcasting mdns advertisement (for port 5002) +236ms
Setting up AirSonos for Living Room {192.168.1.28:1400}
  nodetunes:server starting nodetunes server (Living Room (AirSonos)) +39ms
  nodetunes:server broadcasting mdns advertisement (for port 5003) +235ms
  nodetunes:rtsp received method OPTIONS (CSeq: 0)
{ cseq: '0',
  'x-apple-device-id': '0x649abe03b61f',
  'apple-challenge': 'Bc8pvPQfSobdrzANaYly/w==',
  'dacp-id': '4A1B09E2849756CE',
  'active-remote': '3377123699',
  'user-agent': 'AirPlay/211.3' } +33s
  nodetunes:helper building challenge for Bc8pvPQfSobdrzANaYly/w== (ip: c0a8011e, mac: 5452004cd280) +18ms
  nodetunes:helper computed challenge: CRwjTpAMDIwD/XmpEaQEzSdhROvpxiD9JKHOeqm40I3fj7KfXpLo3HwG+0EpG+nup/RIBolpw+dYfjdtULQWEp/PrnYZbza7LMveR3dfSWr1qZ84tkVTeal7tcw112ctgZx/VfyEPzN5ALzu7GebzTeS6obhWMu2AElOzb+5ZYgVpC2kJEsRjAuBYehSYlKdgVUuSupSDY58/ifgxuS18siXUzFJRBSvE40IpFy9fwAuRcq13oOlfyZNH2CdGtFap3ZLfrLifaINFzxcb/94p7pPblma8h+O2E667jgaJVxefV+pqtlZYViRWKNZsJF2Ng5XTaujTxxf+Fr1jJ50VQ== +11s
  nodetunes:rtsp received method OPTIONS (CSeq: 0)
{ cseq: '0',
  'x-apple-device-id': '0x649abe03b61f',
  'apple-challenge': 'kt5hFM8ZC3GdfrR9MjCWJA==',
  'dacp-id': '91E640F3E9151D9B',
  'active-remote': '1673990108',
  'user-agent': 'AirPlay/211.3' } +11s
  nodetunes:helper building challenge for kt5hFM8ZC3GdfrR9MjCWJA== (ip: c0a8011e, mac: 545200625c26) +8ms
  nodetunes:helper computed challenge: Jd/krI87rQDJGVxTiaNfsRBt1URc4kcNl+EI5YIa2GqH9FV3HkXTJbi3Xo5KGsyD4ULOjp3KnmuYDJgqLZfuAvgbPB+MZFQCmpGQa+8zSJunmnu24K0z2C5GTkhLtl5RT2EWIK6BHd5TEdVM4R5U1zZUAGPM/2hU2SbHtpf37/KOMwRsLCqVO7V/WyBuicwMyeX/nkKDNFMF3Jmhl6HN0fpXjBsvqY3UrzjaoIbGmPPkGSnK3eMN2FEo6SN0e2F0i5/YzH1Eef1PLVajbT7X+/p/HMUgaBId7slCUZYQ3psgziWs1d9CgczdUROqiHygilDq00yQb1BdpCqhlyxN0Q== +13s
  nodetunes:rtsp received method OPTIONS (CSeq: 0)
{ cseq: '0',
  'x-apple-device-id': '0x649abe03b61f',
  'apple-challenge': 'dpw3uodUiruyR/suilyCZw==',
  'dacp-id': 'AAB40522E8C7CA05',
  'active-remote': '1590262237',
  'user-agent': 'AirPlay/211.3' } +21s
  nodetunes:helper building challenge for dpw3uodUiruyR/suilyCZw== (ip: c0a8011e, mac: 5452000d667a) +5ms
  nodetunes:helper computed challenge: ZjKogp7gC0iGHVtfbSGloZOj0sVd5mZqnLpg8Y+vXOSzxzfdtLlFEQ9l0539oHlWm7OcBv5G3W1JL+sU9CqTTX/4zFiXRdZL3/661RZF6LK7jqbNvs/spW9EqO0EfR/GzuiEKFOiaLcRkAsAKaJ5b/2o6ir1NgdbDB0TGCfQzMtMeLRTZFOVkEVmirKIOpnjyGdSWNvsi5Y1Xl2+zKWsUOqVLd2UQGIGOKGY5JmQzr2VdJY4zAbEpOPqZ+bzZg2rd5EZMUWAjd+wPo13oxZrKC860lxHW54/7AcmqowKPKaFGYq5K2aPalX34T7WnWb3khDnG6MCgIEzbPJWx6X4DQ== +13s
  nodetunes:rtsp received method OPTIONS (CSeq: 0)
{ cseq: '0',
  'x-apple-device-id': '0x649abe03b61f',
  'apple-challenge': 'B1eudEuEjkBaXX3SK4jdgg==',
  'dacp-id': 'FB1DAAF14EE8E911',
  'active-remote': '3859770120',
  'user-agent': 'AirPlay/211.3' } +14ms
  nodetunes:helper building challenge for B1eudEuEjkBaXX3SK4jdgg== (ip: c0a8011e, mac: 5452000d667a) +6ms
  nodetunes:helper computed challenge: Y0zyf6QrS16NnHZN9dkw3wADV25KgxgCUOUCaC+oGTmkM/PVaOm3YbBkIhWq0S6JFFlF4Wg5RRif65D5b708Zde2ldG8hOuxhjYdH462yB9GGG0ZWDkZLRVLLRaVxrNTse9yHJgPRlhEU7lIJZ/NEApupG5oXddFFIzOxVlNXLTKtXdHqLNzakx5XQ0OvQaiwUdndmdXbp9oFj1fupo4h4ZTmeAQjI1Z7nBQUMtKm6G3JHpleC2ypCMaJga9i1bYctxsqUZepSDNhvXuX0xW1GTWOuDTeqaLBH7ermZ8xNWSZwrtqB/dLDj7t23FUmmsNLXhSJjlJcs8PijqQCYADg== +12s
  nodetunes:rtsp received method OPTIONS (CSeq: 0)
{ cseq: '0',
  'x-apple-device-id': '0x649abe03b61f',
  'apple-challenge': 'D24sOXdDviN7qXbW1DBtmw==',
  'dacp-id': '98AC3777A655AE73',
  'active-remote': '2081609910',
  'user-agent': 'AirPlay/211.3' } +15ms
  nodetunes:helper building challenge for D24sOXdDviN7qXbW1DBtmw== (ip: c0a8011e, mac: 545200d967b9) +5ms
  nodetunes:helper computed challenge: UgbtBXkI7hu3pHzOCBZ3sgE9FMDOdndcRNb7PTFMFVKeUNF3ET/2ar/jpSp7x6n88XWhFQBPnnYbkRx+f35HP33SbmEVHCkPKMwgy5VUdJ0EcMILzxSgSOx2v9fdQp/htLu6Y4WwO5OaTF8mFf0qntk+nfQZ3A/KNd3cGB8f+OVxTtBDdgVUQOgxvUb//ga71p3xRWPhlxrzMFOBHa/BoG6kRqPTA21HFLcM6ovMv8xiXLDmBxpsXmQAThoDkQcihKgYsTkXuGaZjEHUjRrIEu/jFpqvbe0WXbYHFoQFjrXN8lPrJ8T5gkaz6LLMzhSYIAlsUlgC5HNq228VjSelJQ== +12s
^Cpi@raspberrypi:/tmp$ airsonos --diagnostics
*** WARNING *** The program 'node' uses the Apple Bonjour compatibility layer of Avahi.
*** WARNING *** Please fix your application to use the native API of Avahi!
*** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=node>
*** WARNING *** The program 'node' called 'DNSServiceRegister()' which is not supported (or only supported partially) in the Apple Bonjour compatibility layer of Avahi.
*** WARNING *** Please fix your application to use the native API of Avahi!
*** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=node&f=DNSServiceRegister>
AirSonos Diagnostics
node version	 v0.10.36
operating sys	 linux (arm)
ip address	 192.168.1.30

Searching for Sonos devices on network...

Device 	{"host":"192.168.1.28","port":1400,"state":{}} (ZP120)
`- attrs: 	{"CurrentZoneName":"Living Room",
		"CurrentIcon":"x-rincon-roomicon:living",
		"CurrentConfiguration":"1"}
`- info: 	{"SoftwareVersion":"28.1-83040",
		"DisplaySoftwareVersion":"5.2",
		"HardwareVersion":"1.16.3.1-2",
		"IPAddress":"192.168.1.28",
		"MACAddress":"XX:XX:XX:XX:XX:60",
		"CopyrightInfo":"© 2004-2014 Sonos, Inc. All Rights Reserved.",
		"ExtraInfo":"OTP: 1.1.1(1-16-3-0.9)",
		"HTAudioIn":"0"}
`- topology: 	[{"group":"RINCON_B8E93782F1D401400:8",
		"coordinator":"true",
		"wirelessmode":"1",
		"channelfreq":"2437",
		"behindwifiext":"0",
		"wifienabled":"1",
		"location":"http://192.168.1.27:1400/xml/device_description.xml",
		"version":"28.1-83040",
		"mincompatibleversion":"27.0-00000",
		"legacycompatibleversion":"24.0-0000",
		"bootseq":"5",
		"uuid":"RINCON_B8E93782F1D401400",
		"name":"Portable"},{"group":"RINCON_000E58DF446001400:2",
		"coordinator":"true",
		"wirelessmode":"1",
		"channelfreq":"2437",
		"behindwifiext":"0",
		"wifienabled":"1",
		"location":"http://192.168.1.28:1400/xml/device_description.xml",
		"version":"28.1-83040",
		"mincompatibleversion":"27.0-00000",
		"legacycompatibleversion":"24.0-0000",
		"bootseq":"5",
		"uuid":"RINCON_000E58DF446001400",
		"name":"Living Room"}]


Device 	{"host":"192.168.1.27","port":1400,"state":{}} (ZPS1)
`- attrs: 	{"CurrentZoneName":"Portable",
		"CurrentIcon":"x-rincon-roomicon:portable",
		"CurrentConfiguration":"1"}
`- info: 	{"SoftwareVersion":"28.1-83040",
		"DisplaySoftwareVersion":"5.2",
		"HardwareVersion":"1.8.3.7-2",
		"IPAddress":"192.168.1.27",
		"MACAddress":"XX:XX:XX:XX:XX:D4",
		"CopyrightInfo":"© 2004-2014 Sonos, Inc. All Rights Reserved.",
		"ExtraInfo":"OTP: ",
		"HTAudioIn":"0"}
`- topology: 	[{"group":"RINCON_000E58DF446001400:2",
		"coordinator":"true",
		"wirelessmode":"1",
		"channelfreq":"2437",
		"behindwifiext":"0",
		"wifienabled":"1",
		"location":"http://192.168.1.28:1400/xml/device_description.xml",
		"version":"28.1-83040",
		"mincompatibleversion":"27.0-00000",
		"legacycompatibleversion":"24.0-0000",
		"bootseq":"5",
		"uuid":"RINCON_000E58DF446001400",
		"name":"Living Room"},{"group":"RINCON_B8E93782F1D401400:8",
		"coordinator":"true",
		"wirelessmode":"1",
		"channelfreq":"2437",
		"behindwifiext":"0",
		"wifienabled":"1",
		"location":"http://192.168.1.27:1400/xml/device_description.xml",
		"version":"28.1-83040",
		"mincompatibleversion":"27.0-00000",
		"legacycompatibleversion":"24.0-0000",
		"bootseq":"5",
		"uuid":"RINCON_B8E93782F1D401400",
		"name":"Portable"}]

**Comment by GWHAThistory** [comment]
Same issue here:

'''
Zahavahs-iMac:airsonos zahavahlevine$ sw_vers 
ProductName:	Mac OS X
ProductVersion:	10.10.2
BuildVersion:	14C1514
Zahavahs-iMac:airsonos zahavahlevine$ ifconfig en0
en0: flags=8863<UP,BROADCAST,SMART,RUNNING,SIMPLEX,MULTICAST> mtu 1500
	options=10b<RXCSUM,TXCSUM,VLAN_HWTAGGING,AV>
	ether 3c:07:54:26:cf:cf 
	inet6 fe80::3e07:54ff:fe26:cfcf%en0 prefixlen 64 scopeid 0x4 
	inet6 2601:9:4680:e8e:3e07:54ff:fe26:cfcf prefixlen 64 autoconf 
	inet6 2601:9:4680:e8e:5918:ec0f:7bd5:9448 prefixlen 64 autoconf temporary 
	inet 10.0.1.6 netmask 0xffffff00 broadcast 10.0.1.255
	nd6 options=1<PERFORMNUD>
	media: autoselect (1000baseT <full-duplex,flow-control>)
	status: active
Zahavahs-iMac:airsonos zahavahlevine$ sudo killall coreaudiod
Zahavahs-iMac:airsonos zahavahlevine$ airsonos --diagnostics
AirSonos Diagnostics
node version	 v0.12.1
operating sys	 darwin (x64)
ip address	 10.0.1.6

Searching for Sonos devices on network...

Device 	{"host":"10.0.1.11","port":1400,"state":{}} (ZPS5)
`- attrs: 	{"CurrentZoneName":"Portable",
		"CurrentIcon":"x-rincon-roomicon:portable",
		"CurrentConfiguration":"1"}
`- info: 	{"SoftwareVersion":"29.3-87150-Mainline-Alpha",
		"DisplaySoftwareVersion":"5.4",
		"HardwareVersion":"1.16.4.1-1",
		"IPAddress":"10.0.1.11",
		"MACAddress":"XX:XX:XX:XX:XX:60",
		"CopyrightInfo":"© 2004-2014 Sonos, Inc. All Rights Reserved.",
		"ExtraInfo":"OTP: 1.1.1(1-16-4-zp5s-0.5)",
		"HTAudioIn":"0"}
`- topology: 	[{"group":"RINCON_000E5826805401400:10",
		"coordinator":"true",
		"wirelessmode":"0",
		"hasconfiguredssid":"0",
		"channelfreq":"2412",
		"behindwifiext":"0",
		"wifienabled":"1",
		"location":"http://10.0.1.2:1400/xml/device_description.xml",
		"version":"29.3-87150-Mainline-Alpha",
		"mincompatibleversion":"26.0-00000",
		"legacycompatibleversion":"24.0-0000",
		"bootseq":"39",
		"uuid":"RINCON_000E5826805401400",
		"name":"Rest Of House"},{"group":"RINCON_000E58521F6001400:37",
		"coordinator":"true",
		"wirelessmode":"0",
		"hasconfiguredssid":"0",
		"channelfreq":"2412",
		"behindwifiext":"0",
		"wifienabled":"1",
		"location":"http://10.0.1.11:1400/xml/device_description.xml",
		"version":"29.3-87150-Mainline-Alpha",
		"mincompatibleversion":"26.0-00000",
		"legacycompatibleversion":"24.0-0000",
		"bootseq":"50",
		"uuid":"RINCON_000E58521F6001400",
		"name":"Portable"}]


Device 	{"host":"10.0.1.2","port":1400,"state":{}} (ZP90)
`- attrs: 	{"CurrentZoneName":"Rest Of House",
		"CurrentIcon":"x-rincon-roomicon:living",
		"CurrentConfiguration":"1"}
`- info: 	{"SoftwareVersion":"29.3-87150-Mainline-Alpha",
		"DisplaySoftwareVersion":"5.4",
		"HardwareVersion":"1.1.16.4-1",
		"IPAddress":"10.0.1.2",
		"MACAddress":"XX:XX:XX:XX:XX:54",
		"CopyrightInfo":"© 2004-2014 Sonos, Inc. All Rights Reserved.",
		"ExtraInfo":"r8: 16.0-7",
		"HTAudioIn":"0"}
`- topology: 	[{"group":"RINCON_000E58521F6001400:37",
		"coordinator":"true",
		"wirelessmode":"0",
		"hasconfiguredssid":"0",
		"channelfreq":"2412",
		"behindwifiext":"0",
		"wifienabled":"1",
		"location":"http://10.0.1.11:1400/xml/device_description.xml",
		"version":"29.3-87150-Mainline-Alpha",
		"mincompatibleversion":"26.0-00000",
		"legacycompatibleversion":"24.0-0000",
		"bootseq":"50",
		"uuid":"RINCON_000E58521F6001400",
		"name":"Portable"},{"group":"RINCON_000E5826805401400:10",
		"coordinator":"true",
		"wirelessmode":"0",
		"hasconfiguredssid":"0",
		"channelfreq":"2412",
		"behindwifiext":"0",
		"wifienabled":"1",
		"location":"http://10.0.1.2:1400/xml/device_description.xml",
		"version":"29.3-87150-Mainline-Alpha",
		"mincompatibleversion":"26.0-00000",
		"legacycompatibleversion":"24.0-0000",
		"bootseq":"39",
		"uuid":"RINCON_000E5826805401400",
		"name":"Rest Of House"}]
'''

**Comment by kstokoz** [comment]
I had same problem for whatever reason connect to sonos would fail.

~~~
 nodetunes:helper building challenge for JBFK6YhSchX/AI0YXJXNMQ== (ip: c0a8026a, mac: 54520096c91e) +1ms
  nodetunes:helper computed challenge: Qv1Qz46px5U+jgvCjWqw5OKYBB2tdkto0rpb1ckJ/hoTcT9KF3kv64PAzxNBTpMduOgC4RY6J4Wgb2Bdgx5uPOf7tui0N/5o+zZqS49qNEKpBQ4ory5GIUDtMEOYyp6ymd8rl55HAYMF/GZnLwtIKTEQWwoHFqklXkgCr3j9HS4/kkeDfNA+JIoBO9LB/kj36udQhsJY7dZMal8f1A9weP3xg2noZZipEuS4BBlo6TMG/YpSw9W+UBEJeCd8GhWyiHxnaokQmz1dhXPmxoUjC7+/R6hYgnN/RrrhFfVifT79B9GOYUjwg3mtv46KiQAOZPsifaBmg6bi52WFOSyq9w== +72ms
  nodetunes:rtsp received method ANNOUNCE (CSeq: 1)
~~~

Looking itno this error i thought it can be related that challenge build for another ip than expected.
On my mac i have ipv4 and ipv6 enabled for wifi. 

Disabling ipv6  (`networksetup -setv6off Wi-Fi`) solved issue for me.

**Comment by bronzeiii** [comment]
Disabling ipv6 works for Yosemite OSX.

**Comment by benzwu** [comment]
networksetup -setv6off Wi-Fi

works for me too

**Comment by stephen** [comment]
This is fixed in 0.2.2.
